memory.js

```diff
@@ -0,0 +1,201 @@
+var Game = function(boardElements) {
+	this.boardElements = boardElements;
+	this.tileClass = boardElements.tileHtmlElementClass.substr(1);
+	this.maxValue = boardElements.height*boardElements.width/2;
+	this.takenValues = "#";
+	this.values = null;
+	this.clicked1 = null;
+	this.clicked2 = null;
+	this.remainingMatches = boardElements.height*boardElements.width/2;
+};
+
+Game.prototype.taken = function(value) {
+	var taken = true;
+	var index = this.takenValues.indexOf("#" + value + "#");
+	var lastIndex = this.takenValues.lastIndexOf("#" + value + "#");
+	if ( index === -1 || index === lastIndex ) {
+		taken = false;
+	} 
+	return taken;
+};
+
+Game.prototype.genValue = function() {
+	var found = false;
+	var value = -1;
+	while ( !found ) {
+		value = Math.floor(Math.random()*this.maxValue);
+		if ( !this.taken(value) ) {
+			found = true;
+			this.takenValues += value + "#";
+		}
+	}
+	return value;
+};
+
+Game.prototype.genValues = function() {
+	var values = [];
+	for (var i = 0; i < this.boardElements.height; i++) {
+		values[i] = [];
+		for (var j = 0; j < this.boardElements.width; j++) {
+			values[i][j] = this.genValue();
+		}
+	}
+	return values;
+};
+
+Game.prototype.genHtml = function() {
+	var board = "<table>";
+	for (var i = 0; i < this.boardElements.height; i++) {
+		board += "<tr>";
+		for (var j = 0; j < this.boardElements.width; j++) {
+			board += "<td class=\"" + this.tileClass + "\"><div>" + this.values[i][j] + "</div></td>";
+		}
+		board += "</tr>";
+	}
+	board += "</table>";
+	return board;
+};
+
+Game.prototype.unsetClickedTiles = function() {
+	this.clicked1.hideTile();
+	this.clicked1 = null;
+	this.clicked2.hideTile();
+	this.clicked2 = null;
+};
+
+Game.prototype.removeClickedTiles = function() {
+	this.clicked1.removeTile();
+	this.clicked1 = null;
+	this.clicked2.removeTile();
+	this.clicked2 = null;
+};
+
+Game.prototype.checkWinner = function() {
+	var self = this;
+	if ( --self.remainingMatches === 0 ) {
```

Comment:
While this line works, it would be easier to read if it were split to two lines.